pkg/store/gcs_test.go

```diff
@@ -0,0 +1,116 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/oklog/ulid"
+	"github.com/prometheus/tsdb/fileutil"
+	"google.golang.org/api/iterator"
+
+	"cloud.google.com/go/storage"
+	"github.com/improbable-eng/thanos/pkg/testutil"
+)
+
+func randBucketName(t *testing.T) string {
+	b := make([]byte, 16)
+	_, err := rand.Read(b)
+	testutil.Ok(t, err)
+	return fmt.Sprintf("test-%x", b)
+}
+
+func TestGCSStore_downloadBlocks(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	project, ok := os.LookupEnv("GCP_PROJECT")
+	// TODO(fabxc): make it run against a mock store if no actual bucket is configured.
+	if !ok {
+		return
```

Comment:
you can do t.Skip(...) first

Reply:
Will do in a subsequent PR with another test.